test/Explode.lean

```diff
@@ -0,0 +1,177 @@
+import Mathlib.Tactic.Explode
+import Mathlib.Data.Real.Basic
+set_option linter.unusedVariables false
+open Lean
+
+-- See that the explode command itself works
+#explode true_iff
+
+elab "#explode_test " theoremStx:ident : command => Elab.Command.liftTermElabM do
+  let theoremName : Name ← Elab.resolveGlobalConstNoOverloadWithInfo theoremStx
+  let body : Expr := ((← getEnv).find? theoremStx.getId).get!.value!
+  let entries ← Mathlib.Explode.explode body
+  let md : MessageData ← Mathlib.Explode.entriesToMessageData entries
+  let str := toString (← md.format)
+
+  let some ds ← findDocString? (← getEnv) theoremName
+    | throwError s!"❌ {theoremName}: no docstring"
+  if str == ds then
+    Lean.logInfo s!"✅ {theoremName} success"
+  else
+    Lean.logError s!"❌ {theoremName} in docstring:\n\n{ds}\n❌ {theoremName} actual:\n\n{str}"
+
+/--
+0│   │ a  ├ True
+1│0,0│ ∀I │ True → True
+-/
+theorem lambda : True → True :=
+  λ a => a
+#explode_test lambda
+
+/--
+0│       │ True.intro │ True
+1│_,_,0,0│ @And.intro │ True ∧ True
+-/
+theorem application : True ∧ True :=
+  And.intro True.intro True.intro
+#explode_test application
+
+/--
+0│     │ p  ├ Prop
+1│     │ hP ├ p
+2│0,1,1│ ∀I │ ∀ (p : Prop), p → p
+-/
+theorem theorem_1 : ∀ (p : Prop), p → p :=
+  λ (p : Prop) => (λ hP : p => hP)
+#explode_test theorem_1
+
+/--
+0│         │ p          ├ Prop
+1│         │ q          ├ Prop
+2│         │ hP         ├ p
+3│         │ hQ         ├ q
+4│0,1,2,3  │ @And.intro │ p ∧ q
```

Comment:
IIRC the lean 3 `#explode` would hide non-propositional arguments to functions, to emphasize the logical structure better. Was this behavior removed deliberately?

Reply:
Deliberately, yes, I judged by what makes me understand the fitch table easier. Seeing all arguments to a particular function was helpful, and it doesn't produce too many rows.

Reply:
I think it is somewhat like using `@` on all the functions, I would prefer it to be an option and disabled by default. Especially since most of the subexpressions aren't even numbered, you just get a bunch of `_` most of the time.

Reply:
Yeah I see your point, was on the verge was for this. I can add it when I get some time (next week?), or if you/@kmill could take it up it would also be nice.

Reply:
I've made it omit filtered-out arguments in the deps column, like the mathlib3 version. The exception is that `∀I` lines will still refer to them.  The `_` code is still there in case we eventually want an option to show all arguments to theorems or not.

Regarding `@`, this was purely a side effect of the pretty printer, since the rule is that if a constant's first argument is implicit then you need the `@`.  It took some figuring things out, but I found a way to pretty print the constant itself while still having terminfo.

I also made `#explode` parse more like `#check`, and now it handles arbitrary terms rather than just theorem statements. The command also respects `variable` declarations.